techniques/fileDistribution/fileTemplate/1.0/fileTemplate.ps1.st

```diff
@@ -0,0 +1,95 @@
+  param (
+      [parameter(Mandatory=$true)]  [string]$reportId,
+      [parameter(Mandatory=$true)]  [string]$techniqueName,
+      [switch]$auditOnly
+  )
+    
+    $trackingkey = @(
+    &TRACKINGKEY:{directiveId |
+    "&directiveId&" };separator=","& )
+
+    $sources = @(
+    &FILE_TEMPLATE_TEMPLATE:{source |
+    "&source&" };separator=","& )
+
+    $isRaws = @(
+    &FILE_TEMPLATE_RAW_OR_NOT:{isRaw |
+    "&isRaw&" };separator=","& )
+
+    $rawSources = @(
+    &FILE_TEMPLATE_RAW_TEMPLATE:{rawSource |
+    "&rawSource&" };separator=","& )
+
+    $destinations = @(
+    &FILE_TEMPLATE_AGENT_DESTINATION_PATH:{destination |
+    "&destination&" };separator=","& )
+
+    $posthooks = @(
+    &FILE_TEMPLATE_TEMPLATE_POST_HOOK_COMMAND:{destination |
+    "&destination&" };separator=","& )
+
+    $HashCheckType = "md5"
+    $local_classes = New-ClassContext
+
+    for ($i=0; $i -lt $trackingkey.length; $i++) {
+      $destinationsCanon = Canonify-Class($destinations[$i])
+      $templateDestinations = "C:\Program Files\Rudder\tmp\${destinationsCanon}"
+
+      $componentName = "Templates location"
+      #Check for templates folder
+      $local_classes = Merge-ClassContext $local_classes $(Directory-Create -target $templatesFolder -ReportId $trackingkey[$i] -componentName $componentName -componentKey $destinations[$i] -TechniqueName $techniqueName -auditOnly:$auditOnly -Report:$true)
+      #No need to check for permissions, they are well defined by the inheritence of Rudder folder
+      if ($isRaws[$i] -like "Raw") {
+        #First, write the input in a file
+        $componentName = "Load Template from a file or text input"
+        $local_classes = Merge-ClassContext $local_classes $(File-Enforce-Content -File $templateDestinations -Lines $rawSources[$i] -ReportId $trackingkey[$i] -Enforce $true  -componentName $componentName -componentKey $destinations[$i] -TechniqueName $techniqueName -auditOnly:$auditOnly -Report:$true)
+
+      } else {
+        #First, get the template from the server
+        $componentName = "Load Template from a file or text input"
+        $local_classes = Merge-ClassContext $local_classes $(File-From-Shared-Folder -Source $sources[$i] -Destination $templateDestinations -HashType $HashCheckType -ReportId $trackingkey[$i] -TechniqueName $techniqueName  -componentName $componentName -componentKey $destinations[$i] -auditOnly:$auditOnly -Report:$true)
+      }
+      $componentName = "Expand template"
+      #Expand the template
+      $local_classes = Merge-ClassContext $local_classes $(File-From-Template-Mustache -SourceTemplate $templateDestinations -Destination $destinations[$i] -ReportId $trackingkey[$i] -TechniqueName $techniqueName -componentName $componentName -componentKey $destinations[$i] -auditOnly:$auditOnly -Report:$true)
+
+      #Posthook execution
+      $class_prefix = "file_from_template_${destinationsCanon}_repaired"
+      $state = try {
+        if ($posthooks[$i] -like "") {
```

Comment:
what is the benefit of -like compared to -eq ?

Reply:
None, it is even slower in this case